### DISCUSSION TITLE
Basic commands for the player (play, stop, pause)

### DISCUSSION THREAD

**Discussion by steph-lion** [Androz2091/discord-player#1869] [Q&A] [answered]
Hi. I migrated from an old discord music player to this because this one seems to be up-to-date. I'm honestly struggling on creating basic commands. 
With the old bot, with the `play` command, I had something like
`url -> check user permissions (if it is in voice channel and whatever) -> queue add -> queue play.`
The doc of this module states "_no need to worry about queue management, just use this method 😄_", so I guess that by using `/play <query>` multiple times, it handles the queue automatically.
Now I need to understand how to properly stop the music (so the bot leaves channel), pause it, skip and whatever. In the documentation, the player instance seems to only have `play` as a core command. Am I missing something?
Please post your code examples so I can migrate to this player once for all 😄

**Comment by twlite** [comment, marked as answer]
hello @steph-lion,

Discord Player performs audio player related actions via `queue.node` object, queue related actions via `queue`, etc. Discord Player also has a concept called `hooks` which provide easy access to these objects. If you want to implement a `pause/resume` command for example, you can make the use of `useTimeline` hook:

```js
const { useQueue } = require('discord-player');

const queue = useQueue(interaction.guild.id);

if (!queue) {
  // bot is not in a voice channel
}
if (!queue.currentTrack) {
  // not playing any track
}

// toggle pause action
queue.node.pause();
queue.node.resume();
queue.node.setPaused(true/false);

const state = queue.node.isPaused();

return interaction.reply({ content: `Playback has been ${state ? 'paused' : 'resumed'}` });
```

You can find an example bot [here](https://github.com/Androz2091/discord-player/tree/master/apps/music-bot/src/commands) or [here](https://discord-player.js.org/showcase)
